lcli/src/etl/block_efficiency.rs

```diff
@@ -0,0 +1,317 @@
+    let committee_data = node
+        .get_beacon_states_committees(
+            StateId::Slot(Epoch::start_slot(epoch, T::slots_per_epoch())),
+            None,
+            None,
+            Some(epoch),
+        )
+        .await
+        .map_err(|e| format!("{:?}", e))?
+        .ok_or_else(|| "No committees found".to_string())?
+        .data;
+
+    let committee_info = CommitteeInfo {
+        number_of_committees: committee_data.len(),
+        // FIXME: validators.len() isn't consistent between different committees in the
+        // same epoch. We probably need a better way to do this.
+        // We might be able to just do `active_validator_set.len()`?
+        validators_per_committee: committee_data[0].validators.len(),
+    };
+
+    Ok((committee_data, committee_info))
+}
+
+pub async fn run<T: EthSpec>(matches: &ArgMatches<'_>) -> Result<(), String> {
+    const SECONDS_PER_SLOT: Duration = Duration::from_secs(12);
+    let output_path: PathBuf = clap_utils::parse_required(matches, "output")?;
+    let start_epoch: Epoch = clap_utils::parse_required(matches, "start-epoch")?;
+    let offline_window: u64 = matches
+        .value_of("offline_window")
+        .unwrap_or("3")
+        .parse()
+        .map_err(|e| format!("{:?}", e))?;
+
+    if start_epoch == 0 {
+        return Err("start_epoch cannot be 0.".to_string());
+    }
+    let initialization_epoch: Epoch = start_epoch - 1;
+    let end_epoch: Epoch = clap_utils::parse_required(matches, "end-epoch")?;
+
+    if end_epoch < start_epoch {
+        return Err("start_epoch must be smaller than end_epoch".to_string());
+    }
+
+    let relevant_epochs: Vec<Epoch> = (initialization_epoch.as_u64()..=end_epoch.as_u64())
+        .map(Epoch::new)
+        .collect();
+
+    let mut available_attestations_set: HashSet<UniqueAttestation> = HashSet::new();
+    let mut included_attestations_set: HashMap<UniqueAttestation, InclusionDistance> =
+        HashMap::new();
+
+    // Build validator set HashMap<Index, EpochOfMostRecentAttestation)>
+    // This allows a 'best effort' attempt to normalize block efficiencies.
+    let mut online_validator_set: HashMap<ValidatorIndex, Epoch> = HashMap::new();
+
+    let mut proposer_map: HashMap<Slot, ProposerInfo> = HashMap::new();
+
+    let mut file = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(true)
+        .open(output_path)
+        .map_err(|e| format!("Unable to open file: {}", e))?;
+
+    write!(file, "slot,proposer,available,included,offline,graffiti").unwrap();
+
+    // Initialize API.
+    let endpoint = matches
+        .value_of("endpoint")
+        .unwrap_or("http://localhost:5052/");
+    let node = BeaconNodeHttpClient::new(
+        SensitiveUrl::parse(endpoint).map_err(|_| "Unable to parse endpoint.".to_string())?,
+        Timeouts::set_all(SECONDS_PER_SLOT),
+    );
+
+    // Loop over epochs.
+    for epoch in relevant_epochs {
```

Comment:
```suggestion
    for epoch in (initialization_epoch.as_u64()..=end_epoch.as_u64()).map(Epoch::new) {
```

It's not a big deal, but it does avoid allocating the `relevant_epochs` vec.